### DISCUSSION TITLE
How to run as Kubernetes sidecar?

### DISCUSSION THREAD

**Discussion by kapunakap** [qdm12/gluetun#1054] [Ideas]
The README says `Can work as a Kubernetes sidecar container, thanks @rorph`, but where does it say how?

**Comment by qdm12** [comment]
@rorph would you be able to clarify a bit in this discussion? I will then add it to the wiki. Thanks!!

**Reply by dgseten** [reply]
I'm working also on this topic (as mentioned in https://github.com/qdm12/gluetun/discussions/1148) I will share a functional example as soon as I have something working fine.

**Reply by qdm12** [reply]
Fixed it, it should now exit with `0` when the shutdown is handled gracefully.

**Reply by toniopelo** [reply]
I would be really interested in a explanation of how to make it work as a k8s sidecar as well.
I have two problems as of now :
1. When used as a sidecar, the main container in the k8s pod can't reach the other pods, which is highly problematic (basically, it breaks k8s dns resolution), [see my comment here regarding that matter](https://github.com/qdm12/gluetun/issues/188#issuecomment-1258015354)
2. Gluetun fails to gracefully shut down in my case, and this cause the k8s pod to be marked as errored, and, in my case the associated k8s job to be marked as failed even if the main container did exit gracefully with a 0 exit code, [see my comment here regarding that matter](https://github.com/qdm12/gluetun/discussions/1148#discussioncomment-3787527).

These two things are preventing me to have a functional setup right now, but once I have something that work I'll gladly share my k8s `yaml` files and all to be used as an example if need be @qdm12
@NinoSkopac, @dgseten, did you workaround any of the point I stated above ?
